Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -445,9 +446,10 @@ static MTLVertexStepFunction stepFunction(WGPUVertexStepMode stepMode)
     });
 }
 
-RenderPipeline::RenderPipeline(id<MTLRenderPipelineState> renderPipelineState, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, uint32_t vertexShaderInputBufferCount, Device& device)
+RenderPipeline::RenderPipeline(id<MTLRenderPipelineState> renderPipelineState, MTLRenderPipelineReflection *reflection, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, uint32_t vertexShaderInputBufferCount, Device& device)
```

Comment:
We shouldn't need Metal reflection to build a bind group layout. The rules about how to do it should be rigorously defined in WebGPU + WGSL, not Metal.

Reply:
Yes we can get this information from the WGSL compiler when it is further along. For auto layout I would expect the generated layout from the WGSL compiler to match the reflection information from Metal, otherwise when building the argument buffer the encoding will not match. 

But really this path should only be triggered when the pipeline is created with auto layout. Otherwise we should return the layout passed to create pipeline. I will address that.